pkg/sos/kubernetes.go

```diff
@@ -0,0 +1,75 @@
+package sos
+
+import (
+	"context"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	organizationLabel = "appuio.io/organization"
+	namespaceLabel    = "crossplane.io/claim-namespace"
+)
+
+type ClusterInfo struct {
+	url, token string
+}
+
+type BucketDetail struct {
+	K8sName      string
+	Namespace    string
+	Organization string
+	BucketName   string
+}
+
+func extract(ctx context.Context, clusterInfo ClusterInfo) (*[]BucketDetail, error) {
+	log := ctrl.LoggerFrom(ctx)
+	k8sConfig := rest.Config{Host: clusterInfo.url, BearerToken: clusterInfo.token}
+	k8sClient, err := client.New(&k8sConfig, client.Options{
+		Scheme: runtime.NewScheme(),
```

Comment:
I'm pretty sure you need to add the `exoscalev1.Bucket` CRD to the scheme before the client starts to work properly.

Something like
```
import "github.com/vshn/provider-exoscale/apis"
...
Scheme: apis.AddToScheme(runtime.NewScheme()),
```

see: https://github.com/vshn/provider-exoscale/blob/master/operator_command.go#L93

Reply:
This is resolved.